sound/soc/sof/sof-acpi-dev.c

```diff
@@ -286,14 +267,11 @@ static int sof_acpi_probe(struct platform_device *pdev)
 	dev_dbg(dev, "created machine %s\n",
 		dev_name(&sof_pdata->pdev_mach->dev));
 
-	/* continue SST probing after firmware is loaded */
-	dev_info(dev, "info: loading firmware %s\n", mach->sof_fw_filename);
-	ret = request_firmware_nowait(THIS_MODULE, true, mach->sof_fw_filename,
-				      dev, GFP_KERNEL, priv, sof_acpi_fw_cb);
+	/* register sof-audio platform driver */
+	ret = sof_create_platform_device(priv);
 	if (ret) {
 		platform_device_unregister(sof_pdata->pdev_mach);
-		dev_err(dev, "error: failed to load firmware %s\n",
-			mach->sof_fw_filename);
+		dev_err(dev, "error: failed to create platform device!\n");
```

Comment:
if we fail to create the platform device, does the runtime_pm part make sense or should it be skipped?

Reply:
should be skipped, let me correct it.